src/aktualizr_repo/repo.cc

```diff
@@ -184,7 +196,7 @@ void Repo::readKeys() {
     KeyType key_type;
     key_type_str >> key_type;
     std::string private_key_string(Utils::readFile(p / "private.key"));
-    keys_[Uptane::Role(p.path().filename().string())] =
+    keys_[Uptane::Role(p.path().filename().string(), true)] =
```

Comment:
This is wrong. `delegations` should only be set to `true` if it really is a delegation. I added some logic in my branch to `role.cc` to prevent creating delegations with reserved names, and after rebasing on top of this branch, my check breaks because of this line.

Reply:
Done